### DISCUSSION TITLE
libcurl is crashing due to ssh poll timeout, setting LIBCURL_TIMEOUT is not working as expected.

### DISCUSSION THREAD

**Discussion by anuanton** [curl/curl#15425] [Q&A]
I am using libcurl to do a network transfer. The Curl connection timeout was provided as 300 seconds. The application process has a healthcheck time of 360 seconds. The curl_easy_perfrom was stuck at some point, and the crash was happening at around 11 minutes. The initial piece of code was 

curl_easy_setopt(m_curl_handle,CURLOPT_URL, m_remote_url.c_str());
curl_easy_setopt(m_curl_handle, CURLOPT_POSTQUOTE, headerlist);
curl_easy_setopt(m_curl_handle, CURLOPT_SSL_VERIFYHOST, 0L);
curl_easy_setopt(m_curl_handle, CURLOPT_SSL_VERIFYPEER, 0L);
curl_easy_setopt(m_curl_handle, CURLOPT_READDATA, hd_src);
curl_easy_setopt(m_curl_handle, CURLOPT_INFILESIZE_LARGE, (curl_off_t)fsize);
curl_easy_setopt(m_curl_handle, CURLOPT_CONNECTTIMEOUT,300);
CURLcode result = curl_easy_perform(m_curl_handle);

This program was crashing with the below core dump:
Thread 1 (Thread 0x7fba2cac90c0 (LWP 1981780)):
#0  0x00007fba26fd4ac1 in poll () from /lib64/libc.so.6
#1  0x00007fba27e6f7b8 in ssh_poll_ctx_dopoll () from /lib64/libssh.so.4
#2  0x00007fba27e70890 in ssh_handle_packets () from /lib64/libssh.so.4
#3  0x00007fba27e7095d in ssh_handle_packets_termination () from /lib64/libssh.so.4
#4  0x00007fba27e53d04 in ssh_channel_read_timeout () from /lib64/libssh.so.4
#5  0x00007fba27e7e384 in sftp_packet_read () from /lib64/libssh.so.4
#6  0x00007fba27e7e7b0 in sftp_read_and_dispatch () from /lib64/libssh.so.4
#7  0x00007fba27e82538 in sftp_open () from /lib64/libssh.so.4
#8  0x00007fba280ed2d0 in myssh_statemach_act () from /lib64/libcurl.so.4
#9  0x00007fba280eed0a in myssh_multi_statemach () from /lib64/libcurl.so.4
#10 0x00007fba280e3ab0 in multi_runsingle () from /lib64/libcurl.so.4
#11 0x00007fba280e4b41 in curl_multi_perform () from /lib64/libcurl.so.4
#12 0x00007fba280db42b in curl_easy_perform () from /lib64/libcurl.so.4

From few internet sources, the below settings were added:
curl_easy_setopt(m_curl_handle, CURLOPT_TIMEOUT,302);
curl_easy_setopt(m_curl_handle, CURLOPT_NOSIGNAL,1L);

Still the crash is happening as before. 
Can someone help to understand the reason for crash, and why the timeout functionality is not working?

**Comment by bagder** [comment]
See #15426
